extensions/shared/upgrade-nudge/index.jsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+import { __, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { Button } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import getSiteFragment from '../get-site-fragment';
+import './store';
+
+import './style.scss';
+
+const UpgradeNudge = ( { feature, plan, planName } ) => (
+	<div className="upgrade-nudge">
+		<Gridicon className="upgrade-nudge__icon" icon="star" size={ 18 } />
+		<div className="upgrade-nudge__info">
+			<span className="upgrade-nudge__title">
+				{ sprintf( __( 'Upgrade to %(planName)s', 'jetpack' ), {
+					planName,
+				} ) }
+			</span>
+			<span className="upgrade-nudge__message">
+				{ __( 'To make this block visible on your site', 'jetpack' ) }
+			</span>
+		</div>
+		<Button
+			className="upgrade-nudge__button"
+			href={ addQueryArgs( `https://wordpress.com/plans/${ getSiteFragment() }`, {
+				feature,
```

Comment:
Just noting that plans-grid doesn't get highlighted for `simple-payments` feature. We'll need to check them block-by-block: https://github.com/Automattic/wp-calypso/issues/34399

Reply:
Yep, that's a fun one. It works for WP.com though. Sometimes. (Half-kidding, the mix of `feature`, `plan`, and `customerType` qargs can be... interesting.)

Reply:
The feature actually does get highlighted for .com plans grid, I was accidentally staring at Jetpack plans grid. :-) 

https://github.com/Automattic/wp-calypso/issues/34399#issuecomment-507621247